### DISCUSSION TITLE
architecture/ServiceMesh

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#133] [General]
# architecture/ServiceMesh

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/architecture/ServiceMesh.html

**Comment by raydied** [comment]
在 章节1.5.3 第一小节 1. "...如图 1-23 所示，服务网络由此得名。" ，这句中的服务网络是服务网格吗？
其实，我看图1-23 确实有点像网络。不知道当初为什么叫做service mesh。
不过结合现实世界里 **每个** 社区的网格员，mesh 又比网络恰当。

**Reply by isno** [reply]
是我的错别字，正确的是 “服务网格由此得名”

**Comment by jingchu000** [comment]
Sidecar模式：与Pod一起部署，当集群规模越来越大时，导致延迟问题，和资源占用问题
Proxyless 和 SideCarless 模型：为了不与Pod一起部署，一个使用通信协议解决，一个从内核层面解决，但依然会有一些问题，比如说从内核解决会带来版本，安全等考量。从协议层去解决，一样是需要自己写逻辑处理。
Ambient Mesh：以DaemonSet的形式部署，在4层 和 7层处做处理。 
个人总结，如有不对还请指点
